### PR TITLE
20220405 declaration after statement

### PR DESCRIPTION
**Pull request by douzzer**
fixes various `-Wdeclaration-after-statement` warnings.

tested with `wolfssl-multi-test.sh ... quick-check`.

### CODE REVIEW

File: wolfssl/wolfcrypt/types.h [hunk 1]
```diff
@@ -547,6 +549,7 @@ decouple library dependencies with standard string, memory and so on.
             VAR_TYPE VAR_NAME[VAR_SIZE]
         #define WC_DECLARE_ARRAY(VAR_NAME, VAR_TYPE, VAR_ITEMS, VAR_SIZE, HEAP) \
             VAR_TYPE VAR_NAME[VAR_ITEMS][VAR_SIZE]
+        #define WC_INIT_ARRAY(VAR_NAME, VAR_TYPE, VAR_ITEMS, VAR_SIZE, HEAP) do {} while(0)
```
Comment (dgarske) [inline wolfssl/wolfcrypt/types.h:552]: Is the `do {} while(0)` really needed?

**Review by dgarske — CHANGES_REQUESTED** [review]
Awesome cleanups. I'd be happy to merge it as-is. Left a comment about if we really need the do/while. I haven't seen that as an issue. No don't have to change it.

**Comment by douzzer** [comment]
I too have my doubts about the `do {} while(0)`, but AFAIK it's the only universally safe way to swallow a semicolon.  we use it elsewhere -- although upon review, only John and I do it elsewhere :-)

File: wolfssl/wolfcrypt/types.h (hunk 1 above)
Comment (douzzer) [reply wolfssl/wolfcrypt/types.h:552]: I too have my doubts about the `do {} while(0)`, but AFAIK it's the only universally safe way to swallow a semicolon. we use it elsewhere -- although upon review, only John and I do it elsewhere :-)

**Review by dgarske — APPROVED** [review]

**Pull request merged**